### PR TITLE
Maintenance: Skip tasks based on nx affected graph

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

First iteration of #19880, hopefully allowing us to take a first step into using nx affected, and then spending more time to figure out how to get #19880 in a good state later

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/lib/builder-vite/src/vite-config.ts [hunk 1]
```diff
@@ -24,13 +24,13 @@ export type PluginConfigType = 'build' | 'development';
 const configEnvServe: ConfigEnv = {
   mode: 'development',
   command: 'serve',
-  ssrBuild: false,
+  ssrBuild: true,
```
Comment (valentinpalkovic) [inline code/lib/builder-vite/src/vite-config.ts:27]: This change seems to be unrelated to the other changes.

**Comment by ndelangen** [comment]
@yannbf If you think we should keep this open, then let's re-open.

I think re-doing this work when we're ready for it makes more sense, looking at the merge conflicts.
